chapter-4.qmd

```diff
@@ -0,0 +1,246 @@
+---
+title: "Data transformation"
+---
+
+## Prerequisites
+
+```{r}
+library(nycflights13)
+library(tidyverse)
+#> ── Attaching packages ──────────────────────────────────── tidyverse 1.3.2 ──
+#> ✔ ggplot2 3.4.0           ✔ purrr   1.0.1.9000 
+#> ✔ tibble  3.1.8           ✔ dplyr   1.0.99.9000
+#> ✔ tidyr   1.2.1.9001      ✔ stringr 1.5.0      
+#> ✔ readr   2.1.3           ✔ forcats 0.5.2      
+#> ── Conflicts ─────────────────────────────────────── tidyverse_conflicts() ──
+#> ✖ dplyr::filter() masks stats::filter()
+#> ✖ dplyr::lag()    masks stats::lag()
+```
+
+##Solutions
+
+## 4.2.5 Exercises
+
+1.  
+
+```{=html}
+<!-- -->
+```
+a.  
+
+```{r}
+filter(flights,arr_delay >= 2) |> 
+  arrange(desc(arr_delay))
+```
+
+b.  Flew to houston (IAH or HOU)
+
+```{r}
+flights |>
+  filter(dest %in% c("IAH","HOU"))
+```
+
+c.  
+
+```{r}
+flights |>
+  filter(carrier %in% c("UA","DL","AA"))
+```
+
+d.  
+
+```{r}
+flights |>
+  filter(month %in% c(7,8,9))
+```
+
+e.  
+
+```{r}
+flights |> 
+  filter(dep_delay >= 0,arr_delay < -2)
+```
+
+f.  
+
+```{r}
+flights |> 
+  filter(dep_delay <= -1,arr_delay > 30)
+```
+
+2.  
+
+```{r}
+flights |> 
+  arrange(dep_delay) |> 
+  arrange(time_hour)
+```
+
+3.  
+
+```{r}
+flights |> 
+  arrange(desc(arr_delay))
+```
+
+4.Yes
+
+```{r}
+flights |> 
+  arrange(desc(day)) |> 
+  arrange(desc(month))
```

Comment:
This is not how we can determine that there was a flight on each day. Instead try

```r
flights |> 
  distinct(year, month, day)
```

and check to see if there are 365 unique dates.